backtracking/graph_coloring.cpp

```diff
@@ -1,18 +1,36 @@
+/**
+ * @file
+ * @brief brief one line description here
+ */
 #include <iostream>
 
-/*
- * Number of vertices in the graph
+// created a namespace to ensure all globals and functions are
+// restricted to this file
```

Comment:
Please use a comment block.

Reply:
Dude, this is for you. If I fix the whole code, I will pull to the master branch myself. I am trying to do you a favour and teach you.

Reply:
Sorry about that.
Thanks for this PR, though.

Reply:
study the fixes I made and how I navigated through your questions.. you wanted to know how to get rid of global variables. How did I do it in this code?
How did I add related documetnation, etc.. Would help you to make better contributions and make better fixes.